### ISSUE TITLE
Onedark broken with Vim 8.1.888

### ISSUE THREAD

**Issue by mojoTX** [joshdick/onedark.vim#148]
- [X] I have carefully read and followed the [Installation section of the README](https://github.com/joshdick/onedark.vim#installation)
- [X] I have searched through the [closed issues](https://github.com/joshdick/onedark.vim/issues?q=is%3Aissue+is%3Aclosed) to see if there is an existing solution for my issue

### Terminal Emulator Info

Terminal emulator(s) and their version(s) used, if applicable: `Replace this text if applicable; otherwise delete this section`

### Output From `vim --version`

```
VIM - Vi IMproved 8.1 (2018 May 18, compiled Feb 12 2019 13:34:22)
macOS version
Included patches: 1-888
Compiled by Michael Jarvis <redacted>
Normal version without GUI.  Features included (+) or not (-):
+acl               +extra_search      -mouse_netterm     +tag_old_static
-arabic            -farsi             -mouse_sgr         -tag_any_white
+autocmd           +file_in_path      -mouse_sysmouse    -tcl
-autochdir         +find_in_path      -mouse_urxvt       -termguicolors
-autoservername    +float             +mouse_xterm       -terminal
-balloon_eval      +folding           +multi_byte        +terminfo
-balloon_eval_term -footer            +multi_lang        +termresponse
-browse            +fork()            -mzscheme          +textobjects
+builtin_terms     -gettext           -netbeans_intg     +textprop
+byte_offset       -hangul_input      +num64             +timers
+channel           +iconv             +packages          +title
+cindent           +insert_expand     +path_extra        -toolbar
-clientserver      +job               -perl              +user_commands
+clipboard         +jumplist          +persistent_undo   -vartabs
+cmdline_compl     -keymap            +postscript        +vertsplit
+cmdline_hist      +lambda            +printer           +virtualedit
+cmdline_info      -langmap           -profile           +visual
+comments          +libcall           +python/dyn        +visualextra
-conceal           +linebreak         +python3/dyn       +viminfo
+cryptv            +lispindent        +quickfix          +vreplace
-cscope            +listcmds          +reltime           +wildignore
+cursorbind        +localmap          -rightleft         +wildmenu
+cursorshape       -lua               -ruby              +windows
+dialog_con        +menu              +scrollbind        +writebackup
+diff              +mksession         -signs             -X11
+digraphs          +modify_fname      +smartindent       -xfontset
-dnd               +mouse             +startuptime       -xim
-ebcdic            -mouseshape        +statusline        -xpm
-emacs_tags        -mouse_dec         -sun_workshop      -xsmp
+eval              -mouse_gpm         +syntax            -xterm_clipboard
+ex_extra          -mouse_jsbterm     +tag_binary        -xterm_save
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
  fall-back for $VIM: "/Users/plv8536/my-vim/share/vim"
Compilation: clang -c -I. -Iproto -DHAVE_CONFIG_H   -DMACOS_X -DMACOS_X_DARWIN  -O2               -Wall -Wextra -fstack-protector -fstack-protector-all -pipe -std=c11 -pedantic -arch x86_64 -mtune=native -mmacosx-version-min=10.14 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1
Linking: clang   -L/usr/local/lib -o vim        -lm -lncurses  -liconv -framework AppKit
```

### Issue Description

Vim version 8.1.0888 made changes to the a: dict code, which seems to break this color scheme.

```
commit 31b816042fca879b11965ddd75287732563ba698 (HEAD, tag: v8.1.0888)
Author: Bram Moolenaar <Bram@vim.org>
Date:   Sun Feb 10 22:14:27 2019 +0100

    patch 8.1.0888: the a: dict is not immutable as documented

    Problem:    The a: dict is not immutable as documented.
    Solution:   Make the a:dict immutable, add a test. (Ozaki Kiichi, Yasuhiro
                Matsumoto, closes #3929)
```

Starting up Vim 8.1.888 with `colorscheme onedark` results in many error messages like the following:

I have isolated this to that particular Vim patch, and the onedark colorscheme. 

```
line   13:
E461: Illegal variable name: a:highlight
line   14:
E121: Undefined variable: a:highlight
line   18:
E121: Undefined variable: a:highlight
E116: Invalid arguments for function has_key(a:highlight, "cterm") && a:highlight["cterm"] == "italic"
line   21:
E121: Undefined variable: a:highlight
E116: Invalid arguments for function has_key(a:highlight, "gui") && a:highlight["gui"] == "italic"
line   30:
E121: Undefined variable: a:highlight
E116: Invalid arguments for function has_key(a:highlight, "fg") ? a:highlight.fg.cterm : "NONE")
line   31:
E121: Undefined variable: a:highlight
E116: Invalid arguments for function has_key(a:highlight, "bg") ? a:highlight.bg.cterm : "NONE")
line   34:
E121: Undefined variable: a:highlight
[ etc... ]
```

**Comment by mojoTX** [comment]
For more information, see this Vim closed issue:

https://github.com/vim/vim/pull/3929

**Comment by joshdick** [comment]
Fixed in [04ba479](/joshdick/onedark.vim/commit/04ba47974e4a468e0c1ad32d867594ffc63c159c). Thanks very much for the report!
